scripts/hud/jumpstats.js

```diff
@@ -0,0 +1,133 @@
+		this.addToBuffer(this.strafesBuffer, lastJumpStats.strafeCount);
+		this.addToBuffer(this.syncBuffer, lastJumpStats.strafeSync.toFixed(2));
+		this.addToBuffer(this.gainBuffer, lastJumpStats.speedGain.toFixed(2));
+		this.addToBuffer(this.strafeRatioBuffer, lastJumpStats.strafeRatio.toFixed(2));
+		this.addToBuffer(this.heightDeltaBuffer, lastJumpStats.heightDelta.toFixed(1));
+		this.addToBuffer(this.efficiencyBuffer, lastJumpStats.jumpEfficiency.toFixed(2));
+
+		this.setText();
+	}
+
+	static initializeBuffer(size) {
+		return Array(size).fill('');
+	}
+
+	static addToBuffer(buffer, value) {
+		buffer[buffer.length - 1] += '\n';
+		buffer.push(value);
+
+		if (buffer.length > this.bufferLength) buffer.shift();
+	}
+
+	static onConfigChange() {
+		this.jumpStatsConfig = this.panel.jumpStatsCFG;
+	}
+
+	static onLoad() {
+		this.onConfigChange();
+		this.countBuffer = this.initializeBuffer(this.bufferLength);
+		this.takeoffSpeedBuffer = this.initializeBuffer(this.bufferLength);
+		this.speedDeltaBuffer = this.initializeBuffer(this.bufferLength);
+		this.takeoffTimeBuffer = this.initializeBuffer(this.bufferLength);
+		this.timeDeltaBuffer = this.initializeBuffer(this.bufferLength);
+		this.strafesBuffer = this.initializeBuffer(this.bufferLength);
+		this.syncBuffer = this.initializeBuffer(this.bufferLength);
+		this.gainBuffer = this.initializeBuffer(this.bufferLength);
+		this.strafeRatioBuffer = this.initializeBuffer(this.bufferLength);
+		this.heightDeltaBuffer = this.initializeBuffer(this.bufferLength);
+		this.efficiencyBuffer = this.initializeBuffer(this.bufferLength);
+
+		this.setText();
+	}
+
+	static setText() {
+		this.panel.SetDialogVariable(
+			'jump_count',
+			this.countBuffer.reduce((sum, element) => sum + element)
+		);
+		this.panel.SetDialogVariable(
+			'speed',
+			this.takeoffSpeedBuffer.reduce((sum, element) => sum + element)
+		);
+		this.panel.SetDialogVariable(
+			'speed_delta',
+			this.speedDeltaBuffer.reduce((sum, element) => sum + element)
+		);
+		this.panel.SetDialogVariable(
+			'time',
+			this.takeoffTimeBuffer.reduce((sum, element) => sum + element)
+		);
+		this.panel.SetDialogVariable(
+			'time_delta',
+			this.timeDeltaBuffer.reduce((sum, element) => sum + element)
+		);
+		this.panel.SetDialogVariable(
+			'strafes',
+			this.strafesBuffer.reduce((sum, element) => sum + element)
+		);
+		this.panel.SetDialogVariable(
+			'sync',
+			this.syncBuffer.reduce((sum, element) => sum + element)
+		);
+		this.panel.SetDialogVariable(
+			'gain',
+			this.gainBuffer.reduce((sum, element) => sum + element)
+		);
+		this.panel.SetDialogVariable(
+			'strafe_ratio',
+			this.strafeRatioBuffer.reduce((sum, element) => sum + element)
+		);
+		this.panel.SetDialogVariable(
+			'height_delta',
+			this.heightDeltaBuffer.reduce((sum, element) => sum + element)
+		);
+		this.panel.SetDialogVariable(
+			'efficiency',
+			this.efficiencyBuffer.reduce((sum, element) => sum + element)
+		);
```

Comment:
Worth optimising this a little, since it runs so frequently. Every time you pass an arrow function to `reduce` V8 is going to create a new function (I don't know exactly how this is represented in memory, there's probably some crazy optimisations, but let's clear this up anyway). So we can declare a separate method
```js
sumBuffer(buffer) {
    return buffer.reduce((sum, element) => sum + element);
}
```
FWIW `reduce`s are often slower than a regular imperative for loop, and easier to read. But I think with a super simple case like this it's fine and has the same time complexity.

Reply:
haha, that's how this was originally implemented. This is how things are done in the synchronizer code, I've implemented this function in jump stats now. Good shout